### ISSUE TITLE
Ignore additional fields at any level

### ISSUE THREAD

**Issue by joaodrp** [joaodrp/gelf-formatter#6]
In https://github.com/joaodrp/gelf-formatter/pull/1 a new functionality was added to make it possible to ignore specific attributes at the root level. Example:

```py
attrs = ["secret", "password"]

formatter = GelfFormatter(ignored_attrs=attrs)
# or
formatter.ignored_attrs = attrs

logging.debug("app config", extra={"connection": "local", "secret": "verySecret!", "mysql": {"user": "test", "password": "will_be_logged"}})
```
```json
{"version": "1.1", "short_message": "app config", "timestamp": 1557346554.989846, "level": 6, "host": "my-server", "_connection": "local", "_mysql": {"user": "test", "password": "will_be_logged"}}
```

This issue is to propose the ability of ignoring fields at any depth. In the example above, this would allow us to ignore the `mysql/password` attribute as well.

One way to achieve this is to accept a special syntax on the `ignored_attrs` list, where the depth level is represented by a dot `.`, e.g.:

```py
attrs = ["secret", "mysql.password"]
```
The `format` function would then need to inspect the `extra` values taking into account nested fields.

**Comment by girstenbrei** [comment]
I actually thougth about how to achieve this while implementing #1 and implemented a _flattening_ function for dictionaries, joining levels with a dot. That worked for me, the problem was small enough but it has some obvious problems with name clashing I'm sure you are aware of. Just for completeness:
```python
d = {
    "mysql.password": "a",
    "mysql": {"password": "b"},
}
```
I'm not sure how to solve this, but I at the moment I think that using the current variable type `List[str]` leads to either ambigous naming or complicated specifications (e.g. escaping the `.`) for the `ignored_attrs` variables.
Both of them may be fine to do by a user, but IMHO it is not a solution implementing it in a lib.

Another solution may be switching `ignored_attrs` from type `List[str]` to `List[Union[str, dict]]`. This would preserve the current api but enable the use of unambigous filters:
```python
attrs = ["secret", {"mysql": "password"}]
formatter = GelfFormatter(ignored_attrs=attrs)
logging.debug("app config", extra={"connection": "local", "secret": "verySecret!", "mysql": {"user": "test", "password": "will_be_logged"}})
```
But this will inerhently add complexity on the implementation side.

One more thing: I personally would like to see a recursive filter option. This (I think) may be easier to implement and would lead (at least in these simple examples) to the same outcome.

I'm sorry, this got more complicated than I thought it would 😆
I'm definitely keen on hearing your views on this!

**Comment by girstenbrei** [comment]
Something I haven't quite thought through:
I don't see a use case for filtering non-extra fields. They are either required by GELF or I can just remove them from `allowed_reserved_attrs`. So fields applicable to the filter-problem originate only from the `extra`-keyword when logging. There is already logic checking for extra attributes at [GelfFormatter:144](https://github.com/joaodrp/gelf-formatter/blob/090699edc5939994776ebda579f574656aeb81f3/gelfformatter/formatter.py#L144).
It may be possible to move the filter logic to a separate method and call it there.

**Comment by joaodrp** [comment]
@girstenbrei I also agree that a simple recursive filter option (i.e. ignore all attributes with name `x`at any level) would be the easiest way to start. 

We just have to keep in mind that for some payloads this might be very time consuming (and unwanted). So, if doing this, we should probably expose a config to enable/disable recursive filtering.
